From 006a68873e3acefe5c98a6cc5e1c0d59c6f21879 Mon Sep 17 00:00:00 2001
From: Felix Moessbauer <felix.moessbauer@siemens.com>
Date: Thu, 9 Dec 2021 13:55:27 +0100
Subject: [PATCH 2/5] use native package format

---
 debian/source/format | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/source/format b/debian/source/format
index 163aaf8..89ae9db 100644
--- a/debian/source/format
+++ b/debian/source/format
@@ -1 +1 @@
-3.0 (quilt)
+3.0 (native)
-- 
2.30.2

