--- a/tensorflow/tools/pip_package/setup.py
+++ b/tensorflow/tools/pip_package/setup.py
@@ -78,7 +78,7 @@
     # NOTE: As numpy has releases that break semver guarantees and several other
     # deps depend on numpy without an upper bound, we must install numpy before
     # everything else.
-    'numpy ~= 1.19.2',
+    'numpy ~= 1.19.5',
     # Install other dependencies
     'absl-py ~= 0.10',
     'astunparse ~= 1.6.3',
